sirepo/http_request.py

```diff
@@ -134,16 +134,20 @@ def t(v):
 
 def set_post(data=None):
     """Interface for uri_router"""
-    return _set_flask_g(_POST_ATTR, data)
+    # Always remove data (if there)
+    res = sirepo.srcontext.pop(_POST_ATTR, None)
+    if data is not None:
+        sirepo.srcontext.set(_POST_ATTR, data)
+    return res
 
 
 def set_sim_type(sim_type):
```

Comment:
This should be a contextlib. It should stay in http_request, but be called from uri_router appropriately (perhaps call_api needs to be split into to functions?). http_request should call an internal function that assumes (asserts) that srcontext is already initialized by a prior call. This is similar to set_post, I think. I didn't think about this stuff when designing uri_router. Learning Python still. :(